File: sdk/playwrighttesting/Azure.Developer.MicrosoftPlaywrightTesting/src/TestLogger/Utility/Logger.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;

namespace Azure.Developer.MicrosoftPlaywrightTesting.TestLogger.Utility;

internal enum LogLevel
{
    Debug,
    Info,
    Warning,
    Error
}

internal static class Logger
{
    public static void Log(bool enableConsoleLog, LogLevel level, string message)
    {
        if (enableConsoleLog)
        {
            Console.WriteLine($"{DateTime.Now} [{level}]: {message}");
        }
    }
}

File: sdk/guestconfiguration/Azure.ResourceManager.GuestConfiguration/node_modules.bak/@autorest/openapi-to-typespec/src/utils/logger.ts
import { getSession } from "../autorest-session";

export interface Logger {
  info: (message: string) => void;
  error: (message: string) => void;
  warning: (message: string) => void;
  debug: (message: string) => void;
  verbose: (message: string) => void;
}

export function getLogger(scope: string) {
  const session = getSession();
  const { error, warning, debug, verbose } = session;

  return {
    info: function (message: string) {
      session.info(`${scope}: ${message}`);
    },
    error: (message: string) => error(`${scope}: ${message}`, []),
    warning: (message: string) => warning(`${scope}: ${message}`, []),
    debug: (message: string) => debug(`${scope}: ${message}`),
    verbose: (message: string) => verbose(`${scope}: ${message}`),
  };
}
